### PR TITLE
PR: Some fixes to make Spyder work with the latest changes in PyLSP

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- Request documentation in plain text to the server because it sends it now by default as Markdown, which we can't handle.
- Fix error when showing completion hints due to the previous change in the server as well.
- Fix not showing hovers for objects without docstrings. Signatures that come from the server are now separated from two LF characters, which was causing this problem.
- Update outline if a tree is empty but it has cached items from the server. I discovered this bug while working with Spyder these days and although it's not caused by a server update, I think it's important to fix it because it makes the Outline display files with no symbols when they actually have symbols for it.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @ccordoba12

### CODE REVIEW

File: .github/scripts/install.sh [hunk 1]
```diff
@@ -38,6 +38,9 @@ if [ "$USE_CONDA" = "true" ]; then
     else
         mamba install 'numpy<1.23'
     fi
+
+    # Install docstring-to-markdown until we release PyLSP 1.6.0
+    mamba install docstring-to-markdown
```
Comment (ccordoba12) [inline .github/scripts/install.sh:43]: @dalthviz, this needs to be removed when updating our dependencies before releasing 5.4.0.

File: .github/scripts/install.sh [hunk 2]
```diff
@@ -63,6 +66,8 @@ else
         pip install pyqt5==5.12.* pyqtwebengine==5.12.*
     fi
 
+    # Install docstring-to-markdown until we release PyLSP 1.6.0
+    pip install docstring-to-markdown
```
Comment (ccordoba12) [inline .github/scripts/install.sh:70]: Also this.

File: installers/macOS/req-extras.txt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Spyder extra packages
 autopep8
+docstring-to-markdown
```
Comment (ccordoba12) [inline installers/macOS/req-extras.txt:3]: And this.

**Comment by ccordoba12** [comment]
@spyder-ide/core-developers, after this PR you need to install `docstring-to-markdown` in your development environments. Otherwise Spyder will crash.

**Comment by mrclary** [comment]
> @spyder-ide/core-developers, after this PR you need to install `docstring-to-markdown` in your development environments. Otherwise Spyder will crash.

Under what circumstances with this cause Spyder to crash?
Is this not a Spyder dependency or python-lsp-server dependency?
How is this installed for end users? I only see it added to the macOS installer extra packages.

**Comment by ccordoba12** [comment]
That package is a direct dependency of PyLSP 1.6.0, which I just released. The thing is Spyder imports a little function from a PyLSP module which happens to also import `docstring-to-markdown`, so it'll crash if it's not installed.

**Comment by mrclary** [comment]
> That package is a direct dependency of PyLSP 1.6.0, which I just released. The thing is Spyder imports a little function from a PyLSP module which happens to also import `docstring-to-markdown`, so it'll crash if it's not installed.

Okay. Since it's not an optional dependency (plugin) for python-lsp-server, it does not need to be specified in the macOS extra packages (although there is no harm).

**Comment by ccordoba12** [comment]
Correct. I had to add `docstring-to-markdown` as a requirement before releasing PyLSP 1.6.0 so that the Mac app workflow didn't fail. But @dalthviz will remove its entry tomorrow before releasing 5.4.0.

**Pull request merged**